prow/cmd/sinker/main.go

```diff
@@ -124,6 +126,12 @@ func main() {
 		config:        cfg,
 	}
 
+	// Push metrics to the configured prometheus pushgateway endpoint.
```

Comment:
Pushgateways are unfortunately not the appropriate way to expose metrics from long-lived processes, sorry. I thought the example I gave you didn't use one. Let's just serve the metrics ourselves with an HTTP server like in `prow/cmd/hook/main.go`

Reply:
2 questions:
* do we do the http server only `!o.runOnce`? otherwise, sinker is not a long-lived process, right?
* if we start a http server, which one should go as a routine, current clean loop, or http server? I would put the clean loop in a goroutine.

Reply:
TODO
```
We could also now remove the verbose logging as it will be more useful to see the metrics themselves. still nice to have a log message at the end of a reconciliation loop but we don't necessarily need the metric values in there.
```

Reply:
Yes, we should only serve when we're not running once, and we should use a goroutine for it